packages/node-core/src/indexer/store.service.ts

```diff
@@ -477,6 +498,30 @@ group by
     // Cant throw here because even with historical disabled the current height is used by the store
     return getHistoricalUnit(this.historical, this.blockHeader);
   }
+
+  private async getRewindTimestamp(): Promise<number | undefined> {
+    const rewindTimestampKey = generateRewindTimestampKey(this.subqueryProject.network.chainId);
+    const record = await this.globalDataRepo.findByPk(rewindTimestampKey);
+    if (hasValue(record)) {
+      return record.toJSON().value as number;
+    }
+  }
+
+  private async initChainRewindTimestamp() {
+    if (!this.config.multiChain) return;
+    if ((await this.getRewindTimestamp()) !== undefined) return;
+    const rewindTimestampKey = generateRewindTimestampKey(this.subqueryProject.network.chainId);
+    await this.globalDataRepo.create({key: rewindTimestampKey, value: 0});
```

Comment:
There could be an update between getting and setting meaning that values could be overwritten. It would be good to make this an atomic operation.